env_spec.py

```diff
@@ -177,32 +186,36 @@ def parse(env_spec_text):
 
                 if choices_match:
                     if default_value not in choices or default_value == "":
-                        raise EnvSpecSyntaxError("SYNTAX ERROR: Invalid default value.")
+                        raise EnvSpecSyntaxError(
+                            "SYNTAX ERROR: Invalid default value.", line_number
+                        )
 
                 env_spec_entry["default_value"] = default_value
 
             if not choices_match:
                 if default_value_match:
                     env_spec_type = default_value_match.groups()[0]
                 else:
-                    env_spec_type = line
+                    env_spec_type = env_spec_line
 
                 env_spec_type = env_spec_type.strip()
 
                 if env_spec_type not in valid_types_list:
-                    raise EnvSpecSyntaxError("SYNTAX ERROR: Invalid type.")
+                    raise EnvSpecSyntaxError("SYNTAX ERROR: Invalid type.", line_number)
 
                 env_spec_entry["type"] = env_spec_type
             else:
                 env_spec_entry["type"] = "text"
         else:
-            name = line.strip()
+            name = env_spec_line.strip()
             is_variable_name_valid = re.match(
                 alphanumeric_that_does_not_start_with_digit, name
             )
 
             if not is_variable_name_valid:
-                raise EnvSpecSyntaxError("SYNTAX ERROR: Invalid variable name.")
+                raise EnvSpecSyntaxError(
```

Comment:
Can you change the error message to this please?

```
Invalid variable name; it should contain only latin alphanumeric characters, underscores and not start with a digit.
```

---

Also, why are we validating the variable name in two places?